powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/validator/OperationApproveRequestValidator.java

```diff
@@ -31,8 +32,8 @@ public static String validate(OperationApproveRequest source) {
         if (source == null) {
             return "Operation approve request must not be null";
         }
-        if (source.getApplicationId() == null) {
-            return "Application ID must not be null when creating operation";
+        if (source.getApplicationId() == null || !StringUtils.hasText(source.getApplicationId())) {
```

Comment:
`source.getApplicationId() == null` is already part of `StringUtils`, isn't it?

Reply:
Good catch.